narwhals/dataframe.py

```diff
@@ -599,6 +604,8 @@ def __getitem__(
                 2
             ]
         """
+        if isinstance(item, int):
+            item = [item]
```

Comment:
I'm not wild about using `df[0]` to extract a single row...OK with silently allowing it (and testing it), as scikit-learn use this, but perhaps we can get away with not documenting it to not encourage it 🙈